test.js

```diff
@@ -358,6 +358,83 @@ test('symlink(key, linkname) resolve key path', async function (t) {
   await symlinkAndEntry('\\examples\\more\\h.txt', '/examples/more/h.txt')
 })
 
+test('watch() basic', async function (t) {
+  t.plan(4)
+
+  const { drive } = await testenv(t.teardown)
+  const buf = b4a.from('hi')
+
+  const watcher = drive.watch()
+
+  eventFlush().then(async () => {
+    await drive.put('/a.txt', buf)
+  })
+
+  for await (const [current, previous] of watcher) { // eslint-disable-line no-unreachable-loop
+    t.ok(current instanceof Hyperdrive)
+    t.ok(previous instanceof Hyperdrive)
+    t.is(current.version, 2)
+    t.is(previous.version, 1)
```

Comment:
I was testing something to be sure

Why would `current.get('/a.txt')` return null?

`drive.version` is `2`, same as `current.version`

So `current` should have the new file added, right? Something wrong with the internal snapshots maybe?